src/encoded/static/components/cart/clear.js

```diff
@@ -92,16 +92,17 @@ const mapStateToProps = (state, ownProps) => ({
     user: ownProps.sessionProperties,
 });
 const mapDispatchToProps = (dispatch, ownProps) => ({
-    onClearCartClick: elementAtIds => dispatch(removeMultipleFromCartAndSave(elementAtIds, ownProps.sessionProperties.user, ownProps.sessionProperties.admin, ownProps.fetch)),
+    onClearCartClick: elementAtIds => dispatch(removeMultipleFromCartAndSave(elementAtIds, ownProps.sessionProperties.user, !!(ownProps.session && ownProps.session['auth.userid']), ownProps.fetch)),
```

Comment:
I don't understand why this particular replacement of admin->logged in is negated.

Reply:
oh only admins are allowed to clear>?

Reply:
It’s negated negated, so positive. It’s just a trick to convert something that’s truthy to a real boolean. In this case it probably isn’t required, but I tend to like to have real booleans where booleans are expected — a style thing. I never require this of others, but tend to do that myself.